### DISCUSSION TITLE
How to know if image_data_generator from keras works when splitting?

### DISCUSSION THREAD

**Discussion by vetricko** [teamalgoritma/community#336] [Q&A] [answered]
<img width="680" alt="image" src="https://user-images.githubusercontent.com/65656955/191879397-91ec2d00-6cf4-4f37-86a6-a712823c10c1.png">
After creating a data augmentation with image_data_generator, how does it know if it works?

<img width="726" alt="image" src="https://user-images.githubusercontent.com/65656955/191879598-f2f49276-1b9c-48c9-9a5f-aadebc61769f.png">
when I split the data with flow_images_from_directory the images was found the same amount of original data

Is there a function to view the image_data_generator?

**Comment by NabiilahArdini** [comment, marked as answer]
Dear @vetricko, `image_data_generator()` bekerja dengan melakukan [Image Augmentation](https://medium.com/analytics-vidhya/image-augmentation-9b7be3972e27) pada set data random yang ditarik pada fungsi `flow_images_from_directory()`. Pada fungsi `flow_images_from_directory()` itu terlihat bahwa ia menggunakan image data generator yang sudah dibuat sebelumnya. 

Untuk memastikan fungsi bekerja, bisa dilihat output fungsi tersebut. Contohnya, pada kasus di atas, Bapak sudah menentukan proporsi splitting data:
- pada tahapan `image_data_generator()` dengan parameter `validation_split = 0.2` 
- pada tahapan `flow_images_from_directory()` dengan `subset = "validation"/"training"`
Dari hasil output yang Bapak dapatkan, hasilnya sesuai dengan parameter splitting yang Bapak tentukan. Sehingga dapat diketahui bahwa proses `image_data_generator()` berjalan sempurna.

Semoga informasi ini dapat bermanfaat! Apabila comment ini menjawab pertanyaan Bapak, silahkan klik tombol "Mark as Answer" pada comment ini.
